### ISSUE TITLE
pbsmrtpipe report “Segmentation fault” when run 'pbsmrtpipe.pipelines.sa3_hdfsubread_to_subread'

### ISSUE THREAD

**Issue by yang91** [PacificBiosciences/pbsmrtpipe#473]
Hi guys, I've met a difficult issue for me. Since I have the *.bax.h5 file and I want to analysis these with pbsmrtpipe, I ran the 'pbsmrtpipe.pipelines.sa3_hdfsubread_to_subread' to change these *.bax.h5 files to xml files, then it reported one error:
.................................................................................................................................................................................................
[<TaskChunkedBindingNode-chunk_hdfsubreadset_0 pbcoretools.tasks.h5_subreads_to_subread-1 group cf2642e6-ce21-4c28-9473-6d4622ece5bd>, <TaskChunkedBindingNode-chunk_hdfsubreadset_1 pbcoretools.tasks.h5_subreads_to_subread-2 group cf2642e6-ce21-4c28-9473-6d4622ece5bd>, <TaskChunkedBindingNode-chunk_hdfsubreadset_2 pbcoretools.tasks.h5_subreads_to_subread-3 group cf2642e6-ce21-4c28-9473-6d4622ece5bd>]
Starting worker pbcoretools.tasks.h5_subreads_to_subread-1 (1 workers running, 1 total proc in use)
Starting worker pbcoretools.tasks.h5_subreads_to_subread-2 (2 workers running, 2 total proc in use)
Starting worker pbcoretools.tasks.h5_subreads_to_subread-3 (3 workers running, 3 total proc in use)
Task pbcoretools.tasks.h5_subreads_to_subread-3 Failed task pbcoretools.tasks.h5_subreads_to_subread exit code 139 in 347.10 sec (See file '/home/yangxx/bioPac10/I01-Result/tasks/pbcoretools.tasks.h5_subreads_to_subread-3/stderr'.)
Extracted from stderr
/home/yangxx/bioPac10/I01/1-2K/A01_1/Analysis_Results/m160108_083757_42278_c100925412550000001823200205121610_s1_p0.3.bax.h5
Segmentation fault
Completed running <ResolvedToolContract id:pbcoretools.tasks.h5_subreads_to_subread > exitcode 139 in 337.23 sec.

Workflow status 2/6 completed/total tasks
Shutting down.
Completed execution pbsmrtpipe v0.43.0. Workflow Failed in 368.78 sec (6.15 min)
...................................................................................................................................................................................................................

My command line just like the guidance:

'pbsmrtpipe pipeline-id pbsmrtpipe.pipelines.sa3_hdfsubread_to_subread --preset-xml preset.xml --local-only --force-chunk-mode -e eid_hdfsubread:I01.allBaxFiles.hdfsubreadset.xml'

and the preset.xml was produced by command: 
'pbsmrtpipe show-template-details pbsmrtpipe.pipelines.sa3_ds_isoseq -o preset_tasks.xml'. Have anyone met this issue? How should it be solved? Thanks for any helping and replying.

**Comment by evolvedmicrobe** [comment]
Can you post the full contents of `'/home/yangxx/bioPac10/I01-Result/tasks/pbcoretools.tasks.h5_subreads_to_subread-3/stderr'`?

**Comment by yang91** [comment]
Of course yes!
Here are the content in pbcoretools.tasks.h5_subreads_to_subread-3/stderr:

/home/yangxx/bioPac10/I01/1-2K/A01_1/Analysis_Results/m160108_083757_42278_c100925412550000001823200205121610_s1_p0.3.bax.h5
Segmentation fault
Completed running <ResolvedToolContract id:pbcoretools.tasks.h5_subreads_to_subread > exitcode 139 in 321.81 sec.

**Comment by evolvedmicrobe** [comment]
Looks like you may have hit a segfault in bax2bam.  Try running it directly without the wrapper to confirm:

`bax2bam /home/yangxx/bioPac10/I01/1-2K/A01_1/Analysis_Results/m160108_083757_42278_c100925412550000001823200205121610_s1_p0.3.bax.h5`

**Comment by yang91** [comment]
Yes, you are right. I ran the bax2bam directly, then it reported "Segmentation fault" again. How could it be solved?

**Comment by evolvedmicrobe** [comment]
I would first verify that your file wasn't corrupted during a data transfer.  Make sure that the md5 or other file hash matches between the two.  I would also run the h5check tool to validate the files integrity: `https://www.hdfgroup.org/products/hdf5_tools/h5check.html`  Then let us know if the file does/does not appear to be corrupted.

**Comment by yang91** [comment]
Thanks for your warm-hearted reply!!!!
As you told me, I ran the md5sum the check the file hash and it passed. Then I installed the h5check and ran it the check files integrity, which reported "No non-compliance errors found". So I think there should not have problem with my input files.

**Comment by evolvedmicrobe** [comment]
Eck, okay well if the md5 value returned matches the original files value and h5check doesn't notice any problems, we'll likely have to take a look at the file to diagnose any issues.  Are you willing to share data?  I am going to bring in @PacificBiosciences/techsupport to help coordinate the next steps.

**Comment by pb-jlandolin** [comment]
Hi @yang91,

If you have a PacBio instrument, I can send you some credentials to upload files to us.  I just need to know what university or institute you are from.

Otherwise, can if you have a drop box location we would be happy to download the file to try and reproduce the error.

**Comment by yang91** [comment]
Sorry for my delay reply.  Yes of course, I'd likely to share the data with you. But the data I ran before is too big, so I chosen another much smaller one ran the pbsmrtpipe and got the same issues to send to you.  Could you give me an email or something else to send the data?

**Comment by yang91** [comment]
Ehh.... We don't have any PacBio instrument. I'll upload my data to my drop box and give it out latter.

**Comment by yang91** [comment]
@pb-jlandolin Sorry for my sooooooooo long time delayed. Here are the downlaod links;
h5 file: https://yunpan.cn/cMkcyuSpVM8Hg
log file: https://yunpan.cn/cMkcjiE6bsTez
md5sum file: https://yunpan.cn/cMkcQNVjuUdKh

Please just let me know if you can download these data, thank you !
